### PR TITLE
Open related tickets links in new tab

### PR DESCRIPTION
**Pull request by fvzzy**
✌️ 

/cc @zendesk/vegemite 

### Description

- Fixes bug and opens related ticket link in new ticket tab rather than existing iframe

### CODE REVIEW

File: spec/related_tickets_app.spec.js [hunk 1]
```diff
@@ -4,15 +4,35 @@ import RelatedTicketsApp from '../src/javascripts/related_tickets_app';
...
+    beforeEach(() => {
+      spyOn(client, 'invoke');
+    });
+
+    it('should open a new ticket tab with the correct ticket ID', () => {
+      const relatedTicketLink = document.createElement('a');
+      relatedTicketLink.dataset.ticketId = '123';
+      const e = {
+        target: relatedTicketLink,
+        preventDefault: () => { return null; }
```
Comment (adammw) [inline spec/related_tickets_app.spec.js]: TIL that arrow functions allow you to do this: `() => null`.

File: src/templates/results.hdbs [hunk 2]
```diff
@@ -5,7 +5,7 @@
         {{#each tickets}}
           <tr{{#if ../tooltip_enabled}} class="_tooltip" data-title="{{description}}" data-placement="left" {{/if}}>
             <td>#{{id}}</td>
-            <td><a href="/agent/#/tickets/{{id}}">{{subject}}</a></td>
+            <td><a class="related-ticket-link" data-ticket-id={{id}} href="javascript:;">{{subject}}</a></td>
```
Comment (fvzzy) [inline src/templates/results.hdbs]: Yeah, fair. Wasn't sure the best way to handle this - that seems neater.

File: spec/related_tickets_app.spec.js (hunk 1 above)
Comment (fvzzy) [inline spec/related_tickets_app.spec.js]: @sandlerr could, but we don't really need a spy?

File: spec/related_tickets_app.spec.js [hunk 3]
```diff
@@ -4,15 +4,35 @@ import RelatedTicketsApp from '../src/javascripts/related_tickets_app';
...
+    beforeEach(() => {
+      spyOn(client, 'invoke');
+    });
+
+    it('should open a new ticket tab with the correct ticket ID', () => {
+      const relatedTicketLink = document.createElement('a');
+      relatedTicketLink.dataset.ticketId = '123';
+      const e = {
+        target: relatedTicketLink,
+        preventDefault: () => null
```
Comment (sandlerr) [inline spec/related_tickets_app.spec.js]: I was suggesting `preventDefault: jasmine.createSpy()` if you have jasmine available, or creating a spy using whatever test framework. That will make the intention more clear.

File: spec/related_tickets_app.spec.js (hunk 1 above)
Comment (sandlerr) [inline spec/related_tickets_app.spec.js]: true, you won't verify the spy, but it just shows that you want this property to be callable without caring what it returns.

File: src/javascripts/related_tickets_app.js [hunk 4]
```diff
@@ -97,10 +97,16 @@ const App = {
           ticket.description = ticket.description.substr(0,300).concat("…");
         });
 
-
-        this.switchTo('results', {
-          tickets: tickets,
-          tooltip_enabled: !this.setting('disable_tooltip')
+        client.context().then(data => {
+          const host = data.host;
```
Comment (adammw) [inline src/javascripts/related_tickets_app.js]: haha it sure does look like that, but that wasn't the intention. host was the initial name before product, ie between 'support' and 'chat'. pretty sure we'll always want this to be 'zendesk.com', unless you can see a way of getting the account url.

Comment (adammw) [inline src/javascripts/related_tickets_app.js]: also, should be https

File: spec/related_tickets_app.spec.js [hunk 5]
```diff
@@ -4,15 +4,35 @@ import RelatedTicketsApp from '../src/javascripts/related_tickets_app';
...
 describe('RelatedTicketsApp', () => {
   let app;
+  let client;
 
   beforeEach(() => {
-    const client = ZAFClient.init();
-    const currentUserLocale = 'en-US';
+    client = ZAFClient.init();
 
+    const currentUserLocale = 'en-US';
```
Comment (adammw) [inline spec/related_tickets_app.spec.js]: What about other languages, how will they be loaded?

Comment (fvzzy) [inline spec/related_tickets_app.spec.js]: Does it matter for the purposes of the unit tests? It should be up to the framework to set the correct locale, no?

Comment (adammw) [inline spec/related_tickets_app.spec.js]: 😅  I did not see this was a unit test... carry on!

**Pull request merged**